gap/display.gi

```diff
@@ -7,59 +7,75 @@
 ##
 #############################################################################
 ##
-
+# TO DO Uncomment lines containing StringFormatted when
+# Digraphs requires gap-4.10.0
+# and delete the duplicate code
 # AN's code, adapted by WW
 
-InstallMethod(DotDigraph, "for a digraph by out-neighbours",
-[IsDigraphByOutNeighboursRep],
-function(D)
-  local str, out, i, j;
+BindGlobal("DIGRAPHS_DotDigraph",
+function(D, node_funcs, edge_funcs)
+  local str, out, i, func, j;
   str   := "//dot\n";
   Append(str, "digraph hgn{\n");
   Append(str, "node [shape=circle]\n");
   for i in DigraphVertices(D) do
-    Append(str, Concatenation(String(i), "\n"));
+    # Append(str, StringFormatted("{}", i));
+    Append(str, String(i));
+    for func in node_funcs do
+      Append(str, func(i));
+    od;
+    Append(str, "\n");
   od;
   out := OutNeighbours(D);
   for i in DigraphVertices(D) do
     for j in out[i] do
-      Append(str, Concatenation(String(i), " -> ", String(j), "\n"));
+      # Append(str, StringFormatted("{} -> {}", i, j));
+      Append(str, String(i));
+      Append(str, " -> ");
+      Append(str, String(j));
+      for func in edge_funcs do
+        Append(str, func(i, j));
+      od;
+      Append(str, "\n");
     od;
   od;
   Append(str, "}\n");
   return str;
 end);
 
-InstallMethod(DotVertexLabelledDigraph, "for a digraph by out-neighbours",
+InstallMethod(DotDigraph, "for a digraph by out-neighbours",
 [IsDigraphByOutNeighboursRep],
 function(D)
-  local out, str, i, j;
-  out   := OutNeighbours(D);
-  str   := "//dot\n";
-
-  Append(str, "digraph hgn{\n");
-  Append(str, "node [shape=circle]\n");
-
-  for i in DigraphVertices(D) do
-    Append(str, String(i));
-    Append(str, " [label=\"");
-    Append(str, String(DigraphVertexLabel(D, i)));
-    Append(str, "\"]\n");
-  od;
+  return DIGRAPHS_DotDigraph(D, [], []);
+end);
 
-  for i in DigraphVertices(D) do
-    for j in out[i] do
-      Append(str, Concatenation(String(i), " -> ", String(j), "\n"));
-    od;
-  od;
-  Append(str, "}\n");
-  return str;
+InstallMethod(DotColoredDigraph, "for a digraph by out-neighbours and two lists",
+[IsDigraphByOutNeighboursRep, IsList, IsList],
+function(D, vert, edge)
+  return(DIGRAPHS_DotDigraph(D,
+                             [i -> Concatenation("[color=",
+                             vert[i], ", style=filled]")],
+                             [{i, j} -> Concatenation("[color=",
+                             edge[i][j], "]")]));
+  # return DIGRAPHS_DotDigraph(D,
```

Comment:
Add a comment that the method above can be replaced with the commented out one when/if Digraphs requires GAP version ??.